src/common/ceph_argparse.cc

```diff
@@ -108,12 +108,12 @@ void env_to_vec(std::vector<const char*>& args, const char *name)
     }
     get_str_vec(p, " ", g_str_vec);
   }
-  g_str_vec_lock.unlock();
```

Comment:
@jtlayton can we just drop `g_str_vec_lock`. you added this lock in f72ace0a5666db49bb390ff4ec6808e6e55a779f. and i understand that since `g_str_vec` is a global variable, ideally we should ensure that we need to have a consistent view of this variable using a mutex. but since `clear_g_str_vec()` is only used by tests, and the test of `CephArgParse.env_to_vec` in `ceph_argparse.cc`. and this test is not multithreaded. can we just drop this mutex, for better readability and to simplify the code?